py_gtfs_rt_ingestion/ingest.py

```diff
@@ -110,19 +87,42 @@ def main(event: Dict) -> None:
         config_type = ConfigType.from_filename(files[0])
         converter = get_converter(config_type)
 
-        # filesystem to use when writing parquet files
-        s3_filesystem = fs.S3FileSystem()
-
         for s3_prefix, table in converter.convert(files):
+            # output path is the springboard bucket plus the filetype
             s3_path = os.path.join(export_bucket, s3_prefix)
-            logging.info("Writing Table to %s", s3_path)
-            pq.write_to_dataset(
-                table=table,
-                root_path=s3_path,
-                filesystem=s3_filesystem,
-                partition_cols=converter.partition_cols,
+
+            # generate partitioning for this table write based on what columns
+            # we expect to be able to partition out for this input type
+            partitioning = ds.partitioning(
+                table.select(converter.partition_cols).schema, flavor="hive"
+            )
+
+            # function to run on written files. pyarrow doesn't export type.
+            def post_processing(written_file) -> None:  # type: ignore
+                """
+                file run after each parquet file is wrtten out. log a statent
+                with the filepath and then add it to the metadata table in the
+                performance manager postgres db
+                """
+                logging.info(
+                    "Writing Parquet File. filepath=%s",
+                    written_file.path,
+                )
+                insert_metadata(written_file.path)
+
+            ds.write_dataset(
+                data=table,
+                base_dir=s3_path,
+                filesystem=fs.S3FileSystem(),
+                format=ds.ParquetFileFormat(),
+                partitioning=partitioning,
+                file_visitor=post_processing,
+                basename_template=guid() + "-{i}.parquet",
```

Comment:
Any specific reason for the `guid()` and `-{i}`?

Reply:
this is the default way `write_to_dataset` was generating the basenames for parquet files. if we had a different way we wanted to do this, it would also be fine, but whatever we do would probably be similar.